### PR TITLE
compute merkle root implementation

### PR DESCRIPTION
**Pull request by mubarak23**
- [x] issue #12 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/raito/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

### CODE REVIEW

**Comment by mubarak23** [comment]
@maciejka here is the PR with the unit test for the function

File: src/merkle_tree.cairo [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#[cfg(test)]
+mod tests {
+    use super::{merkle_root};
+    use starknet::core::types::FieldElement;
+
+    #[test]
+    fn test_merkle_root() {
+        let txids = vec![
+            FieldElement::from_hex_be(
+                "50ba87bdd484f07c8c55f76a22982f987c0465fdc345381b4634a70dc0ea0b38"
```
Comment (maciejka) [inline src/merkle_tree.cairo:33]: Why don't you just declare it as `0x50ba87bdd484f07c8c55f76a22982f987c0465fdc345381b4634a70dc0ea0b38_u256`?

File: src/merkle_tree.cairo [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{merkle_root};
+    use starknet::core::types::FieldElement;
+
+    #[test]
+    fn test_merkle_root() {
+        let txids = vec![
```
Comment (maciejka) [inline src/merkle_tree.cairo:31]: This is not cairo. It should rather be `array!`.

**Review by maciejka — CHANGES_REQUESTED** [review]
Please turn this to a pr __into__ pr #43. 
You need to change this line:
image

File: src/merkle_tree.cairo (hunk 1 above)
Comment (mubarak23) [reply src/merkle_tree.cairo:33]: declare it as constant ?

Comment (maciejka) [reply src/merkle_tree.cairo:35]: You should be able to declare u256 Array like:
```
 array![0x50ba87bdd484f07c8c55f76a22982f987c0465fdc345381b4634a70dc0ea0b38_u256, 0x96b8787b1e3abed802cff132c891c2e511edd200b08baa9eb7d8942d7c5423c6_u256,...]
```
using integer literals.

**Review by maciejka — COMMENTED** [review]
Ok, but fix compilation errors.

**Comment by mubarak23** [comment]
> Ok, but fix compilation errors.

that was what i just push

**Review by maciejka — COMMENTED** [review]
Fix tests:
image

**Comment by mubarak23** [comment]
> Fix tests: image

ok

**Comment by maciejka** [comment]
> > Fix tests: image
> 
> ok

@mubarak23 

Please after pushing your commits check on github if all checks pass:
image

You can also run it locally before pushing:
```
scrab build
scarb test
scarb fmt
```

**Comment by mubarak23** [comment]
running scarb build keep returning some error, pointing to this function,

`double_sha256`

currently on it

**Pull request merged**